parsers/nmparser/diagonal_elements.go

```diff
@@ -10,108 +10,312 @@ import "fmt"
 // 2 3
 // there are 3 values and 2 diagonal elements
 var lowerDiagonalLengthToDimension = map[int]int{
-	1:    1,
-	3:    2,
-	6:    3,
-	10:   4,
-	15:   5,
-	21:   6,
-	28:   7,
-	36:   8,
-	45:   9,
-	55:   10,
-	66:   11,
-	78:   12,
-	91:   13,
-	105:  14,
-	120:  15,
-	136:  16,
-	153:  17,
-	171:  18,
-	190:  19,
-	210:  20,
-	231:  21,
-	253:  22,
-	276:  23,
-	300:  24,
-	325:  25,
-	351:  26,
-	378:  27,
-	406:  28,
-	435:  29,
-	465:  30,
-	496:  31,
-	528:  32,
-	561:  33,
-	595:  34,
-	630:  35,
-	666:  36,
-	703:  37,
-	741:  38,
-	780:  39,
-	820:  40,
-	861:  41,
-	903:  42,
-	946:  43,
-	990:  44,
-	1035: 45,
-	1081: 46,
-	1128: 47,
-	1176: 48,
-	1225: 49,
-	1275: 50,
+	0:     0,
```

Comment:
Do these need to be hard-coded like this? Is there a way to just create this map programmatically in the code? If not, I guess this is fine. 

Our upper limit is now 300 ETA's then? @timwaterhouse are we confident that this is more than we would ever realistically need?

Reply:
Never say never! I don't think I've ever had more than about 100 ETAs in a model, but that's not to say it can't be done.  300 is most likely more than enough in the interim, but @dpastoor [mentioned in the issue](https://github.com/metrumresearchgroup/babylon/issues/188#issuecomment-695207331) that this will be derived dynamically in the future.